File: archive/cloudsearch_api_scala/src/main/scala/cloudsearch/search/clients/google/GoogleApis.scala
package cloudsearch.search.clients.google

import cloudsearch.Config
import cloudsearch.db.Account
import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
import com.google.api.client.json.jackson2.JacksonFactory
import com.google.api.services.drive.Drive
import com.google.api.services.gmail.Gmail
import com.google.gdata.client.contacts.ContactsService

/**
  * Created by herval on 7/28/15.
  */
object GoogleApis {
  val applicationName = "cloudsearch-dev"
  val jsonFactory = JacksonFactory.getDefaultInstance
  val httpTranspot = GoogleNetHttpTransport.newTrustedTransport()

  def credential(account: Account) = {
    new GoogleCredential().setAccessToken(account.token)
  }

  def gmailClient(account: Account) = {
    new Gmail.Builder(GoogleApis.httpTranspot, GoogleApis.jsonFactory, credential(account))
        .setApplicationName(GoogleApis.applicationName)
        .build()
  }

  def googleDriveClient(account: Account) = {
    val credential = new GoogleCredential().setAccessToken(account.token)
    new Drive.Builder(GoogleApis.httpTranspot, GoogleApis.jsonFactory, credential)
        .setApplicationName(GoogleApis.applicationName)
        .build()

  }

  def googleContactsClient(account: Account) = {
    val credential = new GoogleCredential().setAccessToken(account.token)
    val client = new ContactsService(GoogleApis.applicationName)
    client.setOAuth2Credentials(credential)
    client
  }

  def googleCredential(account: Account, refreshToken: String) = {
    new GoogleCredential.Builder().
        setJsonFactory(GoogleApis.jsonFactory).
        setTransport(GoogleApis.httpTranspot).
        setClientSecrets(Config.googleClientId, Config.googleClientSecret).
        build().
        setAccessToken(account.token).
        setRefreshToken(refreshToken)
  }

}

File: archive/cloudsearch_kotlin/common/src/main/kotlin/cloudsearch/clients/google/GoogleApis.kt
package cloudsearch.clients.google

import cloudsearch.config.Credentials
import cloudsearch.storage.AccountConfig
import com.google.api.client.googleapis.auth.oauth2.GoogleCredential
import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport
import com.google.api.client.json.jackson2.JacksonFactory
import com.google.api.services.drive.Drive
import com.google.api.services.gmail.Gmail
import com.google.api.services.people.v1.PeopleService


/**
 * Created by hfreire on 1/8/17.
 */
object GoogleApis {
    val jsonFactory = JacksonFactory.getDefaultInstance()
    val httpTranspot = GoogleNetHttpTransport.newTrustedTransport()

    private fun credential(account: AccountConfig): GoogleCredential {
        return GoogleCredential().setAccessToken(token(account))
    }

    fun gmailClient(account: AccountConfig, creds: Credentials): Gmail {
        return Gmail.Builder(httpTranspot, jsonFactory, credential(account))
                .setApplicationName(creds.appName)
                .build()
    }

    fun googleDriveClient(account: AccountConfig, creds: Credentials): Drive {
        val credential = GoogleCredential().setAccessToken(token(account))
        return Drive.Builder(httpTranspot, jsonFactory, credential)
                .setApplicationName(creds.appName)
                .build()

    }

    fun googleContactsClient(account: AccountConfig): PeopleService {
        val credential = GoogleCredential().setAccessToken(token(account))
        val client = PeopleService.Builder(httpTranspot, jsonFactory, credential(account))
                .build()
        return client
    }

    private fun token(config: AccountConfig): String? {
        return config.credentials
    }
}